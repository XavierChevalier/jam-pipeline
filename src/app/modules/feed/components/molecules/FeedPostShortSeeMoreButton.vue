<script setup lang="ts">
  import { mdiNewsOutline } from '@/assets/images/icons/MdiExtra'
  import { mdiChevronDown } from '@mdi/js'
  import { useI18n } from 'vue-i18n'
  import AppIcon from '@/app/components/atoms/AppIcon.vue'

  const { t } = useI18n()
</script>

<template>
  <div class="text-neutral text-center">
    <AppIcon size="sm" :path="mdiNewsOutline" />
    <span class="pl-2 align-middle">{{ t('seeMore') }}</span>
    <div>
      <AppIcon size="sm" :path="mdiChevronDown" />
    </div>
  </div>
</template>

<i18n lang="yaml">
en:
  seeMore: 'See all news'
fr:
  seeMore: 'Voir toutes les news'
</i18n>
