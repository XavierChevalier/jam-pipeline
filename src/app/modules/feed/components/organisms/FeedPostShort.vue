<script setup lang="ts">
  import FeedPostShortSeeMoreButton from '@/app/modules/feed/components/molecules/FeedPostShortSeeMoreButton.vue'
</script>

<template>
  <div class="flex flex-col gap-3 bg-white rounded p-2">
    <slot />
    <FeedPostShortSeeMoreButton />
  </div>
</template>
