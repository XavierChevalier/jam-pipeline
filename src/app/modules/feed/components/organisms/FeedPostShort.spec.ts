import FeedPostShortSeeMoreButton from '../molecules/FeedPostShortSeeMoreButton.vue'
import FeedPostShort from './FeedPostShort.vue'
import * as Stories from '@/app/modules/feed/components/organisms/FeedPostShort.stories'
import { shallowMount } from '@vue/test-utils'
import { StorybookTestsGenerator } from '@/tests/tests-generators/StorybookTestsGenerator'

describe('FeedPostShort', () => {
  StorybookTestsGenerator.fromStoriesExports(Stories).snapshotEachStories()

  it('should contains the button to see all news', () => {
    const wrapper = shallowMount(FeedPostShort)

    expect(
      wrapper.findComponent(FeedPostShortSeeMoreButton).exists()
    ).toBeTrue()
  })
})
