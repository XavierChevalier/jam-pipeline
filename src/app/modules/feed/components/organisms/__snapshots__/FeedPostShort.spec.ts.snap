// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedPostShort stories snapshots [snapshot] should render the "With3Items" component story correctly 1`] = `
<div
  class="flex flex-col gap-3 bg-white rounded p-2"
>
  
  
  <div
    class="flex gap-3 items-center"
  >
    <anonymous-stub
      size="sm"
      url="https://picsum.photos/50"
      username="TheBaronless"
    />
    <span>
      TheBaronless shared a song
    </span>
    <div
      class="flex relative flex-shrink-0 gap-2 items-center ml-auto"
    >
      <div
        class="absolute -left-3 w-0 h-0 rounded-full border-3 border-primary"
      />
      <span
        class="text-neutral"
      >
        less than a minute
      </span>
    </div>
  </div>
  <div
    class="flex gap-3 items-center"
  >
    <anonymous-stub
      size="sm"
      url="https://picsum.photos/50?1"
      username="John Gomm"
    />
    <span>
      John Gomm released a new album
    </span>
    <div
      class="flex relative flex-shrink-0 gap-2 items-center ml-auto"
    >
      <div
        class="absolute -left-3 w-0 h-0 rounded-full border-3 border-primary"
      />
      <span
        class="text-neutral"
      >
        about 1 hour
      </span>
    </div>
  </div>
  <div
    class="flex gap-3 items-center"
  >
    <anonymous-stub
      size="sm"
      url="https://picsum.photos/50?2"
      username="John Gomm"
    />
    <span>
      John Gomm shared a playlist
    </span>
    <div
      class="flex relative flex-shrink-0 gap-2 items-center ml-auto"
    >
      <div
        class="absolute -left-3 w-0 h-0 rounded-full border-3 border-primary"
      />
      <span
        class="text-neutral"
      >
        3 days
      </span>
    </div>
  </div>
  
  
  <anonymous-stub />
</div>
`;
